app/Http/Controllers/UserController.php

```diff
@@ -81,16 +81,16 @@ public function index()
         return view('friend_list', ['friends' => $friends]);
     }
 
-    public function friend_profile(Request $request, $id)
+    public function friend_profile($id)
     {
-        $friend = Friend::where('friend_id', $id)
-            ->where('user_id', Auth::id())->get();
-        return view('friend_profile', ['friends' => $friend]);
+        $user = User::where('id',$id)->first();
```

Comment:
`id` で検索をかける時は`find` を使うと良いです
```suggestion
        $user = User::find($id);
```
参考URL: https://readouble.com/laravel/5.7/ja/eloquent.html